python_modules/dagster/dagster/core/asset_defs/asset_collection.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List, NamedTuple
+
+from ..definitions.executor_definition import ExecutorDefinition
+from ..definitions.job_definition import JobDefinition
+from ..definitions.resource_definition import ResourceDefinition
+from .asset import AssetsDefinition
+from .assets_job import build_assets_job, build_job_from_spec
+
+
+class AssetCollection(
+    NamedTuple(
+        "_AssetCollection",
+        [
+            ("assets", List[AssetsDefinition]),
+            ("resource_defs", Dict[str, ResourceDefinition]),
+            ("inner_job", JobDefinition),
+            ("executor_def", ExecutorDefinition),
+        ],
+    )
+):
+    def __new__(cls, assets, resource_defs, executor_def):
+        return super(AssetCollection, cls).__new__(
+            cls,
+            assets=assets,
+            resource_defs=resource_defs,
+            inner_job=build_assets_job(
+                "__REPOSITORY_MEGA_JOB",
+                assets,
+                resource_defs=resource_defs,
+                executor_def=executor_def,
+            ),
+            executor_def=executor_def,
+        )
+
+    @staticmethod
+    def from_list(assets, resource_defs=None, executor_def=None) -> "AssetCollection":
+        return AssetCollection(
+            assets=assets, resource_defs=resource_defs, executor_def=executor_def
+        )
+
+    @staticmethod
+    def from_package(path, resource_defs=None) -> "AssetCollection":
+        pass
+
+    def execute_in_process(self, instance=None):
+        return self.inner_job.execute_in_process(instance=instance)
+
+    def build_job(self, name=None, subset="*", executor_def=None) -> JobDefinition:
+        return build_job_from_spec(
```

Comment:
If I understand, this will expect the subset query to include be composed of op names. I think we need to expect the query to be composed of asset names.

Reply:
According to the implementation of `@asset` (https://github.com/dagster-io/dagster/blob/0581b8ce49390af8e278d2a8aa72d7959a84ce79/python_modules/dagster/dagster/core/asset_defs/decorators.py#L180), it seems to me that the underlying op's name always matches the asset's name. Do you mean asset key?

Reply:
Or, if we expect this to change in the future, then we should separately have an asset_name field on `AssetsDefinition`. Either way I don't think it's a huge lift.

Reply:
A single op can produce multiple assets - e.g. in the dbt case. The `@multi_asset` decorator enables producing `AssetsDefinition`s that work like this.